pubsub/iterator.go

```diff
@@ -438,6 +439,14 @@ func (it *messageIterator) sendModAck(m map[string]bool, deadline time.Duration)
 				recordStat(it.ctx, ModAckTimeoutCount, 1)
 				return nil
 			default:
+				if err == nil {
+					return nil
+				}
+				// This addresses an error where `context deadline exceeded` errors
+				// not captured by the previous case do not cause fatal errors.
```

Comment:
It might be useful to include a scenario (i.e. the one we think is producing this) where this could happen. If there is a relevant GitHub issue or public thread, that might be useful to include as well. This is a pretty odd scenario and its worth being explicit about it imo.

Reply:
Filed and linked https://github.com/googleapis/google-cloud-go/issues/3060